plugins/sql_loader/tasks.py

```diff
@@ -0,0 +1,244 @@
+    )
+
+    tables_and_columns = db_manager.get_tables_and_columns()
+    db_manager.disconnect()
+    checkbox_list = get_checkbox_list_dict(tables_and_columns)
+
+    task_data.data.update(
+        dict(
+            db_tables_and_columns=tables_and_columns,
+            checkbox_list=checkbox_list,
+            db_type=db_type,
+            db_host=db_host,
+            db_port=db_port,
+            db_user=db_user,
+            db_password=db_password,
+            db_database=db_database,
+        )
+    )
+
+    task_data.save(commit=True)
+
+    return "First step: checked database"
+
+
+def retrieve_params_for_second_task(db_id: int, dumped_schema=None) -> dict:
+    task_data: Optional[ProcessingTask] = ProcessingTask.get_by_id(id_=db_id)
+
+    if task_data is None:
+        msg = f"Could not load task data with id {db_id} to read parameters!"
+        TASK_LOGGER.error(msg)
+        raise KeyError(msg)
+
+    # Previous parameters
+    previous_input = FirstInputParameters(
+        db_enum=DBEnum[task_data.data["db_type"]],
+        **{
+            key: value
+            for key, value in task_data.data.items()
+            if key != "db_type"
+            and key != "db_tables_and_columns"
+            and key != "checkbox_list"
+        },
+    )
+
+    db_enum, db_host, db_port, db_user, db_password, db_database = prep_first_inputs(
+        previous_input
+    )
+    params = dict(
+        db_enum=db_enum,
+        db_host=db_host,
+        db_port=db_port,
+        db_user=db_user,
+        db_password=db_password,
+        db_database=db_database,
+    )
+
+    TASK_LOGGER.info(f"Loaded input from previous step from db: {task_data.data}")
+
+    input_params: SecondInputParameters = SecondInputParametersSchema().loads(
+        task_data.parameters if dumped_schema is None else dumped_schema
+    )
+
+    params.update(input_params.__dict__)
+    params["columns_list"] = ", ".join(json_load(input_params.columns_list))
+
+    TASK_LOGGER.info(f"Loaded input parameters from db: {str(input_params)}")
+
+    return params
+
+
+def second_task_execution(
+    db_enum: DBEnum,
+    db_host: str,
+    db_port: int,
+    db_user: str,
+    db_password: str,
+    db_database: str,
+    custom_query: bool,
+    db_query: str,
+    table_name: str,
+    columns_list: str,
+    id_attribute: str,
+    limit: Optional[int] = None,
+    **kwargs,
+) -> pd.DataFrame:
+    db_database = db_database.removeprefix("file://")
+
+    db_manager, _ = db_enum.get_connected_db_manager(
+        db_host, db_port, db_user, db_password, db_database
+    )
+    df = None
+    if custom_query:
+        table_query = db_query
+    else:
+        table_query = f"SELECT {columns_list} FROM {table_name}"
+
+    print(f"table_query: {table_query}")
+
+    if table_query != "":
+        if columns_list != "":
+            if limit is not None:
+                table_query = f"SELECT * FROM ({table_query}) AS temp LIMIT {limit}"
```

Comment:
I fixed some issues with the preview display, when a custom query is used. I also added a check, to ensure limit is an integer greater than zero. Additionally, I am now dealing with the case of a `;` at the end of a query. I couldn't think of any other failure, concerning the query itself.

Possible issue: The plugin can use databases that are run localy. Maybe this could be a security issue? For example, if the Plugin-Runner runs on a server that contains a private database, then the user can access it (given he has the password).

Reply:
There are also comments. (` -- in sql is the comment marker`) They can be defeated by adding a newline at the end.

Yes, the plugin runner may have access to databases that may not be accessible over internet. This is a potential security risk, but we cannot really account for that in the plugin. But If you want you can either create an option to disallow local databases (localhost or file based) so admins can combat this issue with a configuration or add a warning to the plugin-runner documentation about this and other things to watch out for during deployment.

Reply:
Where may I add such concerns in the plugin-runner documentation?
Where should I add the "no localhost" option? Should the plugin itself contain a config.txt?

Reply:
As this is a plugin specific concern, either add this in the plugin description or in a readme besides the plugin code. You could add a general warning that plugins have access to everything the plugin runner has to the plugin runner documentation or readme, but that should already be kind of obvious...